doc/ipythonconsole.rst

```diff
@@ -2,91 +2,145 @@
+* Display of Matplotlib graphics in Spyder's :doc:`plots` pane, if the :guilabel:`Inline` backend is selected under :menuselection:`Preferences --> IPython console --> Graphics --> Graphics backend`, and inline in the console if :guilabel:`Mute inline plotting` is unchecked under the :guilabel:`Plots` pane's options menu.
+
+.. image:: images/console/console-completion.png
+   :alt: Spyder IPython Console, with a popup list of code completion guesses
+
+For information on the features, commands and capabilities built into IPython itself, see the `IPython documentation`_.
+
+.. _IPython documentation: https://ipython.readthedocs.io/en/stable/overview.html
 
-.. note::
-   If on Windows and connecting to a remote machine over ``ssh``, you'll need to install the ``paramiko`` python package first, *e.g.* with ``conda install paramiko`` if using Anaconda.
 
-To connect to an external kernel,
 
-#. Launch an IPython kernel on the local or remote host if one is not already running.
+================
+Special consoles
+================
 
-   * If using Spyder 3.3.0 or later, you'll need to launch the kernel with ``python -m spyder_kernels.console`` (after you've first installed ``spyder-kernels`` on the host with ``<conda/pip> install spyder-kernels``).
-   * If using a version of Spyder before 3.3.0, ``ipython kernel`` should work to launch the kernel, albeit without certain Spyder-specific features.
+Spyder also supports several types of specialized consoles.
+A `Cython console`_ allows you to use the Cython language to speed up your code and call C functions directly from Python.
+A `Sympy console`_ enables creating and displaying symbolic math expressions right inside Spyder.
+Finally, a `Pylab console`_ loads common Numpy and Matplotlib functions by default; while this is deprecated and not recommended for new code, it can be used for older scripts that need it.
+
+.. _Cython console: https://cython.org/#documentation
+.. _Sympy console: https://docs.sympy.org/latest/index.html
+.. _Pylab console: https://matplotlib.org/faq/usage_faq.html#matplotlib-pyplot-and-pylab-how-are-they-related
+
+.. image:: images/console/console-special.gif
+   :alt: Spyder showing opening a new special Console
+
+
+
+============
+Options menu
+============
+
+The options menu allows you to inspect your current environment variables and the contents of your system's ``PATH`` in Spyder :doc:`variableexplorer` browsers with :guilabel:`Show environment variables` and :guilabel:`Show sys.path contents` respectively, and you can have each kernel display how long it has been running with :guilabel:`Show elapsed time`.
+
+.. image:: images/console/console-options-menu.png
+   :alt: Spyder IPython Console with options menu
+
+You can also change the name of the current :guilabel:`IPython console` tab by simply double-clicking it, or with the :guilabel:`Rename tab` option.
+
+.. image:: images/console/console-rename.gif
```

Comment:
Perhaps it'd be nice to introduce a longer pause before clicking on `Rename tab`? I mean, I think that passes too fast in the gif.

This is not critical (the gif can be understood as it is), so it's fine if you don't want to follow my suggestion.

Reply:
I thought about that too (and the same with the Sympy console GIF), but it was pretty borderline so I didn't mention anything. No strong opinion on this one.

Reply:
Seems fine now.